tests/test_fieldset.py

```diff
@@ -37,6 +37,49 @@ def generate_fieldset_data(xdim, ydim, zdim=1, tdim=1):
     return (data, dimensions)
 
 
+def to_xarray_dataset(data: dict[str, np.array], dimensions: dict[str, np.array]) -> xr.Dataset:
+    assert len(dimensions) in [2, 4], "this function only deals with output from generate_fieldset_data()"
+
+    if len(dimensions) == 4:
+        return xr.Dataset(
+            {
+                "U": (["time", "depth", "lat", "lon"], data["U"]),
+                "V": (["time", "depth", "lat", "lon"], data["V"]),
+            },
+            coords=dimensions,
+        )
+
+    return xr.Dataset(
+        {
+            "U": (["lat", "lon"], data["U"]),
+            "V": (["lat", "lon"], data["V"]),
+        },
+        coords=dimensions,
+    )
+
+
+@pytest.fixture
+def multifile_fieldset(tmp_path):
+    stem = "test_subsets"
+
+    timestamps = np.arange(0, 4, 1) * 86400.0
+    timestamps = np.expand_dims(timestamps, 1)
+
+    ufiles = []
+    vfiles = []
+    for index, timestamp in enumerate(timestamps):
+        data, dimensions = generate_fieldset_data(100, 100)
+        path = tmp_path / f"{stem}_{index}.nc"
+        to_xarray_dataset(data, dimensions).pipe(assign_dataset_timestamp_dim, timestamp).to_netcdf(path)
+        ufiles.append(path)
+        vfiles.append(path)
+
+    files = {"U": ufiles, "V": vfiles}
+    variables = {"U": "U", "V": "V"}
+    dimensions = {"lon": "lon", "lat": "lat"}
+    return FieldSet.from_netcdf(files, variables, dimensions, timestamps=timestamps, allow_time_extrapolation=True)
```

Comment:
Not safer to do `allow_time_extrapolation=False` here? Otherwise w might miss OutOfTimeErrors